### DISCUSSION TITLE
Does redux "unload" the state data from memory that it is not actively using?

### DISCUSSION THREAD

**Discussion by trajano** [reduxjs/redux-toolkit#3967] [Q&A] [answered]
I tried to search for `unload` in this section but didn't find any.

Anyway assuming I have data stored in AsyncStorage with redux-persist that contains many pieces of information and not all of them are used in every screen.  Does RTK unload it's data from memory to the persist store or just keep them in memory for the life of the app?

I read this part about limitations of Redux in [WatermelonDB](https://watermelondb.dev/docs#why-watermelon) which made me think of the question.

> For simple apps, using Redux or MobX with a persistence adapter is the easiest way to go. But when you start scaling to thousands or tens of thousands of database records, your app will now be slow to launch (especially on slower Android devices). Loading a full database into JavaScript is expensive!

**Comment by markerikson** [comment, marked as answer]
No, persistence is about syncing data in the store to somewhere else like `localStorage`.

Redux has the entire current store state in memory, in a single variable. Anything pointed to by that nested object is in memory.
